### DISCUSSION TITLE
Server-side rendering not working In next-url

### DISCUSSION THREAD

**Discussion by AvKat** [urql-graphql/urql#2395] [Q&A] [answered]
Hi
I am new to the graphql ecosystem and have been learning to use it with an Apollo server and a next js app.
I just set up next-url to enable sir but whenever I set `ssr: true` in withUrqlClient, the app crashes, saying "ReactDOMServer does not yet support Suspense."
Help plis

**Comment by SanchezQb** [comment]
If you're using getServerSideProps or getStaticProps, you may want to set ssr to false

`export default withUrqlClient(
    ssr => ({
        url: 'http://localhost:4000/graphql',
    }),
    { ssr: false }
)(Index);
`

and then getServerSideProps or getStaticProps, you pass false as the second argument to initUrqlClient

`
     export const getServerSideProps: GetServerSideProps = async (context) => {
    const ssrCache = ssrExchange({ isClient: false });

    const client = initUrqlClient({
        url: 'http://localhost:4000/graphql',
        exchanges: [dedupExchange, cacheExchange, ssrCache, fetchExchange],
    }, false)

    await client?.query(TODOS_QUERY, { id: context.params?.box }).toPromise();

    return {
        props: {
            // urqlState is a keyword here so withUrqlClient can pick it up.
            urqlState: ssrCache.extractData(),
        },
    };
}
`

**Comment by kitten** [comment, marked as answer]
This seems to have been a regression in `next`, see: https://github.com/FormidableLabs/urql/issues/2400
